README.md
<img align="center" src="src/main/assets/media/northem-light-java-banner.png" alt="Northem Light Java"/> <a href="https://github.com/arcticicestudio/northem-light"><img align="right" src="https://raw.githubusercontent.com/arcticicestudio/northem-light-java/develop/src/main/assets/media/northem-logo.png"/></a>

<img src="https://cdn.travis-ci.org/images/favicon-c566132d45ab1a9bcae64d8d90e4378a.svg" width=24 height=24/> [![Travis CI](https://img.shields.io/travis/arcticicestudio/northem-light-java/develop.svg)](https://travis-ci.org/arcticicestudio/northem-light-java) <img src="https://codecov.io/favicon.ico" width=24 height=24/> [![codecov](https://codecov.io/gh/arcticicestudio/northem-light-java/branch/develop/graph/badge.svg)](https://codecov.io/gh/arcticicestudio/northem-light-java)  
<img src="https://assets-cdn.github.com/favicon.ico" width=24 height=24/> [![release](https://img.shields.io/github/release/arcticicestudio/northem-light-java.svg)](https://github.com/arcticicestudio/northem-light-java/releases/latest) <img src="http://search.maven.org/favicon.ico" width=24 height=24/> [![Central Repository](https://img.shields.io/maven-central/v/com.arcticicestudio/northem-light-java.svg)](http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.arcticicestudio%22%20AND%20a%3A%22northem-light-java%22) [![OSS Sonatype Snapshot](https://img.shields.io/badge/snapshot-0.3.0--SNAPSHOT-blue.svg)](https://oss.sonatype.org/content/repositories/snapshots/com/arcticicestudio/northem-light-java)

Java library implementation of the north-bluish light color palette [Northem Light](https://github.com/arcticicestudio/northem-light).  
Optimized for syntax highlighting- and UI projects.  
This is a light and more color-intensive variation.

## Getting started
### Setup
To use northem-light-java it must be available on your classpath.  
You can get it from the <img src="https://search.maven.org/favicon.ico" width=16 height=16/> <a href="https://search.maven.org">Central Repository</a> as a dependency for your favorite build tool or [download the latest version](https://github.com/arcticicestudio/northem-light-java/releases/latest).

<img src="http://apache.org/favicons/favicon.ico" width=16 height=16/> <a href="https://maven.apache.org">Apache Maven</a>
```xml
<dependency>
  <groupId>com.arcticicestudio</groupId>
  <artifactId>northem-light-java</artifactId>
  <version>0.2.0</version>
</dependency>
```

<img src="https://gradle.org/wp-content/uploads/fbrfg/favicon.ico" width=16 height=16/> <a href="https://gradle.org">Gradle</a>
```java
compile(group: 'com.arcticicestudio', name: 'northem-light-java', version: '0.2.0')
```

<img src="http://apache.org/favicons/favicon.ico" width=16 height=16/> <a href="https://ant.apache.org/ivy">Apache Ivy</a>
```xml
<dependency org="com.arcticicestudio" name="northem-light-java" rev="0.2.0" />
```

Development snapshots are available via [OSS Sonatype](https://oss.sonatype.org/content/repositories/snapshots/com/arcticicestudio/northem-light-java).   

### Build
Build and install northem-light-java into your local repository without GPG signing:  
```
mvn clean install
```

Signed artifacts may be build by using the `sign-gpg` profile with a provided `gpg.keyname` property:
```
mvn clean install -Dgpg.keyname=YourGPGKeyId
```

A continuous integration build is running at [Travis-CI](https://travis-ci.org/arcticicestudio/northem-light-java).

### Variations
  - <img src="https://java.com/favicon.ico" width=16 height=16 /> <a href="https://github.com/arcticicestudio/northem-java">Northem - Java</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-java"><img src="https://java.com/favicon.ico" width=16 height=16> Northem Dark - Java</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />

### Other Projects
  - <a href="https://github.com/arcticicestudio/northem-light-intellij-idea-syntax"><img src="https://www.jetbrains.com/_assets//shared/favicons/jetbrains.ico" width=16 height=16> Northem Light IntelliJ IDEA Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-atom-syntax"><img src="https://atom.io/favicon.ico" width=16 height=16> Northem Light Atom Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-atom-ui"><img src="https://atom.io/favicon.ico" width=16 height=16> Northem Light Atom UI</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-eclipse-syntax"><img src="https://eclipse.org/favicon.ico" width=16 height=16> Northem Light Eclipse Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-gedit"><img src="https://static.gnome.org/wiki.gnome.org/gnome/css/favicon.png" width=16 height=16> Northem Light gedit</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 />
  - <a href="#"><img src="http://www.gtk.org/images/gtk-logo.ico" width=16 height=16> Northem Light GTK</a> (in drafting) <img src="https://www.kernel.org/theme/images/logos/favicon.png" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-notepadplusplus"><img src="http://notepad-plus-plus.org/assets/images/favicon.ico" width=16 height=16> Northem Light Notepad++</a> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" width=16 height=16 />

Visit the [official website](http://arcticicestudio.com/northem) for more information.

## Development
[![Changelog](https://img.shields.io/badge/Changelog-v0.2.0-blue.svg)](https://github.com/arcticicestudio/northem-light-java/blob/v0.2.0/CHANGELOG.md)

### Workflow
This project follows the [gitflow](http://nvie.com/posts/a-successful-git-branching-model) branching model.

### Specifications
This project follows the [Arctic Versioning Specification (ArcVer)](https://github.com/arcticicestudio/arcver).

### Contribution
Please report issues/bugs, suggestions for improvements and feature requests to the [issuetracker](https://github.com/arcticicestudio/northem-light-java/issues).

---

<img src="http://arcticicestudio.com/favicon.ico" width=16 height=16/> Copyright &copy; 2016 Arctic Ice Studio

[![GPL-3.0](http://www.gnu.org/graphics/gplv3-88x31.png)](http://www.gnu.org/licenses/gpl.txt) [![CC BY-SA-4.0](http://mirrors.creativecommons.org/presskit/buttons/88x31/svg/by-sa.svg)](http://creativecommons.org/licenses/by-sa/4.0/)


CHANGELOG.md
<img align="center" src="https://raw.githubusercontent.com/arcticicestudio/northem-light-java/develop/src/main/assets/media/northem-light-java-banner.png" alt="Northem Light Java"/> <a href="https://github.com/arcticicestudio/northem-light"><img align="right" src="https://raw.githubusercontent.com/arcticicestudio/northem-light-java/develop/src/main/assets/media/northem-logo.png"/></a>

## 0.2.0 (2016-08-03) - Apache Maven Migration
This version is mainly focused on a [Apache Maven](https://maven.apache.org) migration.
Release and snapshot versions are now deployed to the [Central Repository](https://search.maven.org) via [OSS Sonatype](https://oss.sonatype.org).

### Improvements
#### Build Tool
  - Migrated to [Apache Maven](https://maven.apache.org) based on the [`glacier-apache-maven@0.4.0`](https://github.com/arcticicestudio/glacier-apache-maven) project skeletons

#### Documentation
  - The `README` has been modernized and now includes a "Getting started" section

## 0.1.0 (2016-04-09)
### Features
  - Implemented the [Northem Light](https://github.com/arcticicestudio/northem-light) color palette enumeration class [`NorthemLight`](https://github.com/arcticicestudio/northem-light-java/blob/master/src/main/java/com/arcticicestudio/northem/light/NorthemLight.java)  

**Constants**

| Name         | RGB Value     |
| ------------ | ------------- |
| `DARK_GRAY`  | `69,69,70`    |
| `DARK_GRAY`  | `69,69,70`    |
| `GRAY`       | `109,109,111` |
| `LIGHT_GRAY` | `198,198,200` |
| `DARK_SNOW`  | `242,242,247` |
| `DIRTY_SNOW` | `249,249,252` |
| `SNOW`       | `255,255,255` |
| `AQUA`       | `122,202,194` |
| `LIGHT_BLUE` | `111,186,208` |
| `BLUE`       | `113,167,202` |
| `DARK_BLUE`  | `87,129,179`  |
| `RED`        | `223,92,90`   |
| `YELLOW`     | `255,197,104` |
| `ORANGE`     | `241,144,84`  |
| `LIME`       | `197,201,97`  |
| `PURPLE`     | `186,142,196` |

**Methods**

| Name | Description |
| ---- | ----------- |
| `NorthemLight(final int RED, final int GREEN, final int BLUE)` | Constructs a new color object. |
| `public static String rgb(final Color COLOR)` | Converts the color to the RGB identifier. |
| `public static String hex(final Color COLOR)` | Converts the color to the HEX identifier. |
| `public Color get()` | Returns the color object. |

## 0.0.0 (2016-04-09) - Repository Reinitialization


src/test/java/com/arcticicestudio/northem/light/NorthemLightTest.java
/*
++++++++++++++++++++++++++++++++++++++++++++++++
title     Northem Light Enumeration JUnit Test +
project   northem-light-java                   +
file      NorthemLightTest.java                +
version   0.2.0                                +
author    Arctic Ice Studio                    +
email     development@arcticicestudio.com      +
website   http://arcticicestudio.com           +
copyright Copyright (C) 2016                   +
created   2016-01-24 19:59 UTC+0100            +
modified  2016-04-09 10:57 UTC+0200            +
++++++++++++++++++++++++++++++++++++++++++++++++

[Description]
JUnit test class for the 'NorthemLight' class of the "Northem Light - Java" API.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem Light
  (https://github.com/arcticicestudio/northem-light)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/

package com.arcticicestudio.northem.light;

import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * JUnit test class for the {@link NorthemLight} class of the
 * <a href="https://github.com/arcticicestudio/northem-light-java">Northem Light - Java</a> API.
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see NorthemLight
 * @see <a href="http://junit.org/">http://junit.org/</a>
 * @see <a href="https://github.com/arcticicestudio/northem-light-java">Northem Light - Java</a>
 * @since 0.1.0
 */
public class NorthemLightTest {
  @Test
  public void shouldReturnTheCorrectHEXColorCode() {
    assertEquals(NorthemLight.hex(NorthemLight.DARK_GRAY.get()), "#454546");
    assertEquals(NorthemLight.hex(NorthemLight.GRAY.get()), "#6D6D6F");
    assertEquals(NorthemLight.hex(NorthemLight.LIGHT_GRAY.get()), "#C6C6C8");
    assertEquals(NorthemLight.hex(NorthemLight.DARK_SNOW.get()), "#F2F2F7");
    assertEquals(NorthemLight.hex(NorthemLight.DIRTY_SNOW.get()), "#F9F9FC");
    assertEquals(NorthemLight.hex(NorthemLight.SNOW.get()), "#FFFFFF");
    assertEquals(NorthemLight.hex(NorthemLight.AQUA.get()), "#7ACAC2");
    assertEquals(NorthemLight.hex(NorthemLight.LIGHT_BLUE.get()), "#6FBAD0");
    assertEquals(NorthemLight.hex(NorthemLight.BLUE.get()), "#71A7CA");
    assertEquals(NorthemLight.hex(NorthemLight.DARK_BLUE.get()), "#5781B3");
    assertEquals(NorthemLight.hex(NorthemLight.RED.get()), "#DF5C5A");
    assertEquals(NorthemLight.hex(NorthemLight.YELLOW.get()), "#FFC568");
    assertEquals(NorthemLight.hex(NorthemLight.ORANGE.get()), "#F19054");
    assertEquals(NorthemLight.hex(NorthemLight.LIME.get()), "#C5C961");
    assertEquals(NorthemLight.hex(NorthemLight.PURPLE.get()), "#BA8EC4");
  }

  @Test
  public void shouldReturnTheCorrectRGBColorCode() {
    assertEquals(NorthemLight.rgb(NorthemLight.DARK_GRAY.get()), "rgb(69, 69, 70)");
    assertEquals(NorthemLight.rgb(NorthemLight.GRAY.get()), "rgb(109, 109, 111)");
    assertEquals(NorthemLight.rgb(NorthemLight.LIGHT_GRAY.get()), "rgb(198, 198, 200)");
    assertEquals(NorthemLight.rgb(NorthemLight.DARK_SNOW.get()), "rgb(242, 242, 247)");
    assertEquals(NorthemLight.rgb(NorthemLight.DIRTY_SNOW.get()), "rgb(249, 249, 252)");
    assertEquals(NorthemLight.rgb(NorthemLight.SNOW.get()), "rgb(255, 255, 255)");
    assertEquals(NorthemLight.rgb(NorthemLight.AQUA.get()), "rgb(122, 202, 194)");
    assertEquals(NorthemLight.rgb(NorthemLight.LIGHT_BLUE.get()), "rgb(111, 186, 208)");
    assertEquals(NorthemLight.rgb(NorthemLight.BLUE.get()), "rgb(113, 167, 202)");
    assertEquals(NorthemLight.rgb(NorthemLight.DARK_BLUE.get()), "rgb(87, 129, 179)");
    assertEquals(NorthemLight.rgb(NorthemLight.RED.get()), "rgb(223, 92, 90)");
    assertEquals(NorthemLight.rgb(NorthemLight.YELLOW.get()), "rgb(255, 197, 104)");
    assertEquals(NorthemLight.rgb(NorthemLight.ORANGE.get()), "rgb(241, 144, 84)");
    assertEquals(NorthemLight.rgb(NorthemLight.LIME.get()), "rgb(197, 201, 97)");
    assertEquals(NorthemLight.rgb(NorthemLight.PURPLE.get()), "rgb(186, 142, 196)");
  }
}


src/main/java/com/arcticicestudio/northem/light/package-info.java
/*
+++++++++++++++++++++++++++++++++++++++++++
title     Package Info                    +
project   northem-light-java              +
file      NorthemLight.java               +
version   0.2.0                           +
author    Arctic Ice Studio               +
email     development@arcticicestudio.com +
website   http://arcticicestudio.com      +
copyright Copyright (C) 2016              +
created   2016-01-24 19:47 UTC+0100       +
modified  2016-04-09 10:25 UTC+0200       +
+++++++++++++++++++++++++++++++++++++++++++

[Description]
Root package of the "Northem Light - Java" project, which implements the "Northem Light" color palette as enumeration.
The package represents the public API.
The main entry point of the package is the "com.arcticicestudio.northem.light.NorthemLight" class.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem Light
  (https://github.com/arcticicestudio/northem-light)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/
/**
 * Root package of the <a href="https://github.com/arcticicestudio/northem-light-java">Northem Light - Java</a>
 * project, which implements the <a href="https://github.com/arcticicestudio/northem-light">Northem Light</a>
 * color palette as enumeration.
 * <p>
 *   The package represents the public API.
 *   The main entry point of the package is the {@link com.arcticicestudio.northem.light.NorthemLight} class.
 * </p>
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see <a href="https://github.com/arcticicestudio/northem-light-java">Northem Light - Java</a>
 * @see <a href="https://github.com/arcticicestudio/northem-light">Northem Light</a>
 * @version 0.2.0
 * @since 0.1.0
 */
package com.arcticicestudio.northem.light;


src/main/java/com/arcticicestudio/northem/light/NorthemLight.java
/*
+++++++++++++++++++++++++++++++++++++++++++
title     NorthemLight                    +
project   northem-light-java              +
file      NorthemLight.java               +
version   0.2.0                           +
author    Arctic Ice Studio               +
email     development@arcticicestudio.com +
website   http://arcticicestudio.com      +
copyright Copyright (C) 2016              +
created   2016-01-24 19:46 UTC+0100       +
modified  2016-04-09 10:47 UTC+0200       +
+++++++++++++++++++++++++++++++++++++++++++

[Description]
"Northem Light" color palette as enumeration for the "Northem Light - Java" project.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem Light
  (https://github.com/arcticicestudio/northem-light)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/
package com.arcticicestudio.northem.light;

import javafx.scene.paint.Color;
import java.util.regex.Pattern;

/**
 * <a href="https://github.com/arcticicestudio/northem-light">Northem Light</a> color palette
  * as enumeration of the <a href="https://github.com/arcticicestudio/northem-light-java">Northem Light - Java</a> project.
 * <div>
 *   <table style="display:inline-block">
 *     <thead>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(69,69,70);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(109,109,111);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(198,198,200);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(242,242,247);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(249,249,252);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(255,255,255);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(122,202,194);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(111,186,208);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(113,167,202);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(87,129,179);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(223,92,90);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(255,197,104);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(241,144,84);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(197,201,97);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(186,142,196);margin: 0"></th>
 *     </thead>
 *   </table>
 * </div>
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see <a href="https://github.com/arcticicestudio/northem-light">"Northem Light" GitHub Repository</a>
 * @see <a href="https://github.com/arcticicestudio/northem-light-java">"Northem Light - Java" GitHub Repository</a>
 * @since 0.1.0
 */
public enum NorthemLight {
  /**
   * The color {@code dark-gray} with an RGB value of {@code rgb(69,69,70)} and HEX value of {@code #454546}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(69,69,70);margin: 0"></div>
   */
  DARK_GRAY(69,69,70),
  /**
   * The color {@code gray} with an RGB value of {@code rgb(109,109,111)} and HEX value of {@code #6D6D6F}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(109,109,111);margin: 0"></div>
   */
  GRAY(109,109,111),
  /**
   * The color {@code light-gray} with an RGB value of {@code rgb(198,198,200)} and HEX value of {@code #C6C6C8}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(198,198,200);margin: 0"></div>
   */
  LIGHT_GRAY(198,198,200),
  /**
   * The color {@code dark-snow} with an RGB value of {@code rgb(242,242,247)} and HEX value of {@code #F2F2F7}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(242,242,247);margin: 0"></div>
   */
  DARK_SNOW(242,242,247),
  /**
   * The color {@code dirty-snow} with an RGB value of {@code rgb(249,249,252)} and HEX value of {@code #F9F9FC}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(249,249,252);margin: 0"></div>
   */
  DIRTY_SNOW(249,249,252),
  /**
   * The color {@code snow} with an RGB value of {@code rgb(255,255,255)} and HEX value of {@code #FFFFFF}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(255,255,255);margin: 0"></div>
   */
  SNOW(255,255,255),
  /**
   * The color {@code aqua} with an RGB value of {@code rgb(122,202,194)} and HEX value of {@code #7ACAC2}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(122,202,194);margin: 0"></div>
   */
  AQUA(122,202,194),
  /**
   * The color {@code light-blue} with an RGB value of {@code rgb(111,186,208)} and HEX value of {@code #6FBAD0}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(111,186,208);margin: 0"></div>
   */
  LIGHT_BLUE(111,186,208),
  /**
   * The color {@code blue} with an RGB value of {@code rgb(113,167,202)} and HEX value of {@code #71A7CA}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(113,167,202);margin: 0"></div>
   */
  BLUE(113,167,202),
  /**
   * The color {@code dark-blue} with an RGB value of {@code rgb(87,129,179)} and HEX value of {@code #5781B3}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(87,129,179);margin: 0"></div>
   */
  DARK_BLUE(87,129,179),
  /**
   * The color {@code red} with an RGB value of {@code rgb(223,92,90)} and HEX value of {@code #DF5C5A}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(223,92,90);margin: 0"></div>
   */
  RED(223,92,90),
  /**
   * The color {@code yellow} with an RGB value of {@code rgb(255,197,104)} and HEX value of {@code #FFC568}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(255,197,104);margin: 0"></div>
   */
  YELLOW(255,197,104),
  /**
   * The color {@code orange} with an RGB value of {@code rgb(241,144,84)} and HEX value of {@code #F19054}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(241,144,84);margin: 0"></div>
   */
  ORANGE(241,144,84),
  /**
   * The color {@code lime} with an RGB value of {@code rgb(197,201,97)} and HEX value of {@code #C5C961}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(197,201,97);margin: 0"></div>
   */
  LIME(197,201,97),
  /**
   * The color {@code purple} with an RGB value of {@code rgb(186,142,196)} and HEX value of {@code #BA8EC4}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(186,142,196);margin: 0"></div>
   */
  PURPLE(186,142,196);

  /**
   * Compiled pattern to convert the RGB and HEX color values.
   */
  private static final Pattern COLOR_PREFIX = Pattern.compile("0x", Pattern.LITERAL);

  /**
   * Contains the RGB color code.
   */
  private final Color COLOR;

  /**
   * Constructs a new color object.
   *
   * @param RED The value for the additive primary color {@code red}
   * @param GREEN The value for the additive primary color {@code green}
   * @param BLUE The value for the additive primary color {@code blue}
   */
  NorthemLight(final int RED, final int GREEN, final int BLUE) {
    COLOR = Color.rgb(RED, GREEN, BLUE);
  }

  /**
   * Converts the color to the {@code RGB} identifier.
   *
   * @param COLOR The color object
   * @return a string representation of the {@code RGB} identifier
   */
  public static String rgb(final Color COLOR) {
    String hex = COLOR_PREFIX.matcher(COLOR.toString()).replaceAll("");
    String hexRed = hex.substring(0, 2).toUpperCase();
    String hexGreen = hex.substring(2, 4).toUpperCase();
    String hexBlue = hex.substring(4, 6).toUpperCase();

    String intRed = Integer.toString(Integer.parseInt(hexRed, 16));
    String intGreen = Integer.toString(Integer.parseInt(hexGreen, 16));
    String intBlue = Integer.toString(Integer.parseInt(hexBlue, 16));

    return String.join("", "rgb(", intRed, ", ", intGreen, ", ", intBlue, ")");
  }

  /**
   * Converts the color to the {@code HEX} identifier.
   *
   * @param COLOR The color object
   * @return a string representation of the {@code HEX} identifier
   */
  public static String hex(final Color COLOR) {
    return COLOR_PREFIX.matcher(COLOR.toString()).replaceAll("#").substring(0, 7).toUpperCase();
  }

  /**
   * Returns the color object.
   *
   * @return the color object
   */
  public Color get() {
    return COLOR;
  }
}
